File: src/main/java/fr/vlegall/sochief/model/recipe/RecipeTag.java
package fr.vlegall.sochief.model.recipe;

import com.fasterxml.jackson.annotation.JsonIgnore;
import fr.vlegall.sochief.model.common.AuditableEntity;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(
        name = "RECIPE_TAG",
        uniqueConstraints = {@UniqueConstraint(columnNames = {"recipe_id", "tag_id"})}
)
public class RecipeTag extends AuditableEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "recipe_id", nullable = false)
    @JsonIgnore
    private Recipe recipe;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "tag_id", nullable = false)
    private Tag tag;
}

File: src/main/kotlin/fr/vlegall/sochief/model/recipe/RecipeTag.kt
package fr.vlegall.sochief.model.recipe

import com.fasterxml.jackson.annotation.JsonIgnore
import fr.vlegall.sochief.model.common.AuditableEntity
import jakarta.persistence.*
import jakarta.validation.constraints.NotNull

@Entity
@Table(
    name = "RECIPE_TAG",
    uniqueConstraints = [UniqueConstraint(columnNames = ["recipe_id", "tag_id"])]
)
class RecipeTag(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @field:NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "recipe_id", nullable = false)
    @JsonIgnore
    var recipe: Recipe? = null,

    @field:NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "tag_id", nullable = false)
    var tag: Tag? = null,
) : AuditableEntity()